### PR TITLE
feat(replays): Add replay-video endpoints

### PR DESCRIPTION
**Pull request by cmanallen**
Adds endpoints for downloading replay-video blob data.  Depends on https://github.com/getsentry/sentry/pull/65531.

### CODE REVIEW

**Review by JoshFerge — COMMENTED** [review]
do we need a video details endpoint? i feel like we never use the segment details endpoint, only the segment index, so just wondering if it makes sense to not include

**Comment by cmanallen** [comment]
@JoshFerge The client needs higher throughput from the recording-segments endpoint because it needs all of the segments to render the timeline.  For video we only need one file to render the preview.

We may add an index page later.  We'll see what usage is like.  Right now though I'm not sure how to encode a video collection in a single response and the client is only demanding the one endpoint.

File: src/sentry/replays/endpoints/project_replay_video_details.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        if not segment:
+            return self.respond({"detail": "Replay recording segment not found."}, status=404)
+
+        video = download_video(segment)
+        if video is None:
+            return self.respond({"detail": "Replay recording segment not found."}, status=404)
+
+        video_io = BytesIO(video)
+        response = StreamingHttpResponse(
+            iter(lambda: video_io.read(4096), b""), content_type="application/json"
```
Comment (JoshFerge) [inline src/sentry/replays/endpoints/project_replay_video_details.py:64]: is application/json the correct content type? it seems like below, the test data is not json

Comment (cmanallen) [reply src/sentry/replays/endpoints/project_replay_video_details.py:64]: Oops. Yeah. I refactored and forgot to update this.  Thanks.

Comment (JoshFerge) [reply src/sentry/replays/endpoints/project_replay_video_details.py:64]: what does the second parameter in read do? i don't see it in the docs.

**Review by JoshFerge — APPROVED** [review]

File: src/sentry/replays/endpoints/project_replay_video_details.py (hunk 1 above)
Comment (cmanallen) [reply src/sentry/replays/endpoints/project_replay_video_details.py:64]: The second argument is on the `iter` function.  Its a sentinel value.  It basically means iterate until you see this and then stop.  So basically we'll send 4KB chunks of data until the data runs out and the process can exit.

Comment (JoshFerge) [reply src/sentry/replays/endpoints/project_replay_video_details.py]: ah, right, thanks!

**Pull request merged**